application-templates/starter/src/components/channel-details/channel-details.js

```diff
@@ -0,0 +1,139 @@
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import {
+  formatLocalizedString,
+  transformLocalizedFieldToLocalizedString,
+} from '@commercetools-frontend/l10n';
+import { DOMAINS, NO_VALUE_FALLBACK } from '@commercetools-frontend/constants';
+import { useIsAuthorized } from '@commercetools-frontend/permissions';
+import {
+  useShowNotification,
+  useShowApiErrorNotification,
+} from '@commercetools-frontend/actions-global';
+import { PERMISSIONS } from '../../constants';
+import {
+  useChannelDetailsUpdater,
+  useChannelDetailsFetcher,
+} from '../../hooks/use-channels-connector';
+import { docToFormValues, formValuesToDoc } from './conversions';
+import ChannelsDetailsForm from './channel-details-form';
+import { transformErrors } from './transform-errors';
+import messages from './messages';
+
+const ChannelDetails = (props) => {
+  const intl = useIntl();
+  const params = useParams();
+  const { loading, error, channel } = useChannelDetailsFetcher(params.id);
+  const { dataLocale, projectLanguages } = useApplicationContext((context) => ({
+    dataLocale: context.dataLocale,
+    projectLanguages: context.project.languages,
+  }));
+  const canManage = useIsAuthorized({
+    demandedPermissions: [PERMISSIONS.Manage],
+  });
+  const showNotification = useShowNotification();
+  const showApiErrorNotification = useShowApiErrorNotification();
+  const channelDetailsUpdater = useChannelDetailsUpdater();
+  const handleSubmit = useCallback(
+    async (formikValues, formikHelpers) => {
+      const data = formValuesToDoc(formikValues);
+      try {
+        await channelDetailsUpdater.execute({
+          originalDraft: channel,
+          nextDraft: data,
+        });
+        showNotification({
+          kind: 'success',
+          domain: DOMAINS.SIDE,
+          text: intl.formatMessage(messages.channelUpdated, {
+            channelName: formatLocalizedString(formikValues, {
+              key: 'name',
+              locale: dataLocale,
+              fallbackOrder: projectLanguages,
+            }),
+          }),
+        });
+      } catch (graphQLErrors) {
+        const transformedErrors = transformErrors(graphQLErrors);
+        if (transformedErrors.unmappedErrors.length > 0)
+          showApiErrorNotification({
+            errors: transformedErrors.unmappedErrors,
+          });
+
+        formikHelpers.setErrors(transformedErrors.formErrors);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [channelDetailsUpdater]
+  );
+
+  return (
+    <ChannelsDetailsForm
+      initialValues={docToFormValues(channel, projectLanguages)}
+      onSubmit={handleSubmit}
+    >
+      {(formProps) => {
+        return (
+          <FormModalPage
+            title={formatLocalizedString(
+              {
+                name: transformLocalizedFieldToLocalizedString(
+                  channel?.nameAllLocales ?? []
+                ),
```

Comment:
If you expose `formik.values` from the `ChannelsDetailsForm`, you can use them here instead of duplicating this transformation.

```suggestion
                name: fomProps.values.name,
```